### PR TITLE
Upgrade to react-native 0.71.8

### PR DESCRIPTION
**Pull request by gusgard**
# Why

React Native 0.71.8 https://github.com/facebook/react-native/releases/tag/v0.71.8

`v0.71.8` fixes this issue with `onChangeText`
> Do not send extra onChangeText even when instantiating multiline TextView ([a804c0f22b](https://github.com/facebook/react-native/commit/a804c0f22b4b11b3d9632dc59a6da14f6c4325e3) by [@dmytrorykun](https://github.com/dmytrorykun))

# How

* update react-native@0.71.8 in package.json

# Test Plan

ci?

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by gusgard** [comment]
I couldn’t find a README to guide me through the migration process. Do you know, @Kudo, if I missed any steps?

**Comment by Kudo** [comment]
that's fantastic! i didn't expect that will have contributors create react-native upgrade pr for us 🔥
let's see how ci going. if there are something missed, do you mind me to update your branch directly and work together with you?

**Comment by gusgard** [comment]
> that's fantastic! i didn't expect that will have contributors create react-native upgrade pr for us 🔥
> let's see how ci going. if there are something missed, do you mind me to update your branch directly and work together with you?

Yes, go ahead 👍

**Review by Kudo — COMMENTED** [review]
forget me for previous message because i don't have edit permission for your repo/branch.

could you please help to update these changes?

1. one more missed package.json update:

```diff
diff --git a/apps/sandbox/package.json b/apps/sandbox/package.json
index 8d042f60dc..07765ed6b1 100644
--- a/apps/sandbox/package.json
+++ b/apps/sandbox/package.json
@@ -12,7 +12,7 @@
   "dependencies": {
     "expo": "~48.0.0-beta.0",
     "react": "18.2.0",
-    "react-native": "0.71.7"
+    "react-native": "0.71.8"
   },
   "devDependencies": {
     "babel-preset-expo": "~9.3.0",
```

2. React-Codegen

```diff
diff --git a/ios/versioned-react-native/ABI48_0_0/ReactNative/codegen/ios/ABI48_0_0React-Codegen.podspec.json b/ios/versioned-react-native/ABI48_0_0/ReactNative/codegen/ios/ABI48_0_0React-Codegen.podspec.json
index bd32ec473a..10d1f8e7bb 100644
--- a/ios/versioned-react-native/ABI48_0_0/ReactNative/codegen/ios/ABI48_0_0React-Codegen.podspec.json
+++ b/ios/versioned-react-native/ABI48_0_0/ReactNative/codegen/ios/ABI48_0_0React-Codegen.podspec.json
@@ -1 +1 @@
-{"name":"ABI48_0_0React-Codegen","version":"0.71.7","summary":"Temp pod for generated files for React Native","homepage":"https://facebook.com/","license":"Unlicense","authors":"Facebook","compiler_flags":"-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32 -Wno-documentation -Wno-nullability-completeness -std=c++17","source":{"git":""},"header_mappings_dir":"./","platforms":{"ios":"11.0"},"source_files":"**/*.{h,mm,cpp}","pod_target_xcconfig":{"HEADER_SEARCH_PATHS":"\"$(PODS_ROOT)/boost\" \"$(PODS_ROOT)/RCT-Folly\" \"${PODS_ROOT}/Headers/Public/ABI48_0_0React-Codegen/react/renderer/components\" \"$(PODS_ROOT)/Headers/Private/ABI48_0_0React-Fabric\" \"$(PODS_ROOT)/Headers/Private/ABI48_0_0React-RCTFabric\""},"dependencies":{"ABI48_0_0FBReactNativeSpec":[],"ABI48_0_0React-jsiexecutor":[],"RCT-Folly":[],"ABI48_0_0RCTRequired":[],"ABI48_0_0RCTTypeSafety":[],"ABI48_0_0React-Core":[],"ABI48_0_0React-jsi":[],"ABI48_0_0ReactCommon/turbomodule/bridging":[],"ABI48_0_0ReactCommon/turbomodule/core":[],"ABI48_0_0hermes-engine":[]}}
\ No newline at end of file
+{"name":"ABI48_0_0React-Codegen","version":"0.71.8","summary":"Temp pod for generated files for React Native","homepage":"https://facebook.com/","license":"Unlicense","authors":"Facebook","compiler_flags":"-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32 -Wno-documentation -Wno-nullability-completeness -std=c++17","source":{"git":""},"header_mappings_dir":"./","platforms":{"ios":"11.0"},"source_files":"**/*.{h,mm,cpp}","pod_target_xcconfig":{"HEADER_SEARCH_PATHS":"\"$(PODS_ROOT)/boost\" \"$(PODS_ROOT)/RCT-Folly\" \"${PODS_ROOT}/Headers/Public/ABI48_0_0React-Codegen/react/renderer/components\" \"$(PODS_ROOT)/Headers/Private/ABI48_0_0React-Fabric\" \"$(PODS_ROOT)/Headers/Private/ABI48_0_0React-RCTFabric\""},"dependencies":{"ABI48_0_0FBReactNativeSpec":[],"ABI48_0_0React-jsiexecutor":[],"RCT-Folly":[],"ABI48_0_0RCTRequired":[],"ABI48_0_0RCTTypeSafety":[],"ABI48_0_0React-Core":[],"ABI48_0_0React-jsi":[],"ABI48_0_0ReactCommon/turbomodule/bridging":[],"ABI48_0_0ReactCommon/turbomodule/core":[],"ABI48_0_0hermes-engine":[]}}
```

3. backport textinput fix

```diff
diff --git a/ios/versioned-react-native/ABI48_0_0/ReactNative/Libraries/Text/TextInput/ABI48_0_0RCTBackedTextInputDelegateAdapter.m b/ios/versioned-react-native/ABI48_0_0/ReactNative/Libraries/Text/TextInput/ABI48_0_0RCTBackedTextInputDelegateAdapter.m
index 0e3a464ea5..328ab500a1 100644
--- a/ios/versioned-react-native/ABI48_0_0/ReactNative/Libraries/Text/TextInput/ABI48_0_0RCTBackedTextInputDelegateAdapter.m
+++ b/ios/versioned-react-native/ABI48_0_0/ReactNative/Libraries/Text/TextInput/ABI48_0_0RCTBackedTextInputDelegateAdapter.m
@@ -256,21 +256,21 @@ static void *TextFieldSelectionObservingContext = &TextFieldSelectionObservingCo

 - (void)textViewDidChange:(__unused UITextView *)textView
 {
-  if (_ignoreNextTextInputCall && [_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText]) {
+  if (_ignoreNextTextInputCall) {
     _ignoreNextTextInputCall = NO;
     return;
   }
-  _lastStringStateWasUpdatedWith = _backedTextInputView.attributedText;
   _textDidChangeIsComing = NO;
   [_backedTextInputView.textInputDelegate textInputDidChange];
 }

 - (void)textViewDidChangeSelection:(__unused UITextView *)textView
 {
-  if (![_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText]) {
+  if (_lastStringStateWasUpdatedWith && ![_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText]) {
     [self textViewDidChange:_backedTextInputView];
     _ignoreNextTextInputCall = YES;
   }
+  _lastStringStateWasUpdatedWith = _backedTextInputView.attributedText;
   [self textViewProbablyDidChangeSelection];
 }
```

File: ios/Podfile.lock [hunk 1]
```diff
@@ -4884,4 +4884,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 6bb5b123ef65907681848e92e63a6f9535528e03
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.12.0
```
[the remark this answers is not in the document]
Comment (tsapeta) [reply ios/Podfile.lock:4887]: Afaik it should be `1.12.1` 😉

[the remark this answers is not in the document]
Comment (Kudo) [reply ios/Podfile.lock:4887]: > Afaik it should be `1.12.1` 😉

the base of this pr is sdk-48, so it's 1.11.2 still 😅

**Comment by gusgard** [comment]
> forget me for previous message because i don't have edit permission for your repo/branch.

Added you to my fork https://github.com/gusgard/expo

> could you please help to update these changes?

Done

---

@Kudo  Which command did you run to generate the changes in the `codegen` folders? I did it manually

File: packages/expo/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,12 @@
 
 ### 💡 Others
 
+## 48.0.17 — 2023-05-11
+
```
Comment (tsapeta) [inline packages/expo/CHANGELOG.md:14]: The version and date will be added automatically by the publish script 😉 
```suggestion
```

File: packages/expo-dev-menu/CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
 ### 📚 3rd party library updates
 
 - Update `react-native` to 0.71.7. ([#22253](https://github.com/expo/expo/pull/22253) by [@kudo](https://github.com/kudo))
+- Update `react-native` to 0.71.8. ([#22458](https://github.com/expo/expo/pull/22458) by [@gusgard](https://github.com/gusgard))
```
Comment (tsapeta) [inline packages/expo-dev-menu/CHANGELOG.md:16]:
```suggestion
- Update `react-native` to 0.71.8. ([#22253](https://github.com/expo/expo/pull/22253) by [@kudo](https://github.com/kudo), [#22458](https://github.com/expo/expo/pull/22458) by [@gusgard](https://github.com/gusgard))
```

File: packages/expo-dev-launcher/CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@
 ### 📚 3rd party library updates
 
 - Update `react-native` to 0.71.7. ([#22253](https://github.com/expo/expo/pull/22253) by [@kudo](https://github.com/kudo))
+- Update `react-native` to 0.71.8. ([#22458](https://github.com/expo/expo/pull/22458) by [@gusgard](https://github.com/gusgard))
```
Comment (tsapeta) [inline packages/expo-dev-launcher/CHANGELOG.md:16]:
```suggestion
- Update `react-native` to 0.71.8. ([#22253](https://github.com/expo/expo/pull/22253) by [@kudo](https://github.com/kudo), [#22458](https://github.com/expo/expo/pull/22458) by [@gusgard](https://github.com/gusgard))
```

**Review by tsapeta — APPROVED** [review]

File: packages/expo/CHANGELOG.md (hunk 2 above)
Comment (gusgard) [reply packages/expo/CHANGELOG.md:14]: Thanks 👍

**Review by Kudo — APPROVED** [review]
thanks for updating all the changes. now it looks good. let's merge it and i will publish new updates this week.

**Pull request merged**